storage-model/src/main/java/no/unit/nva/publication/storage/model/daos/WithCristinIdentifier.java

```diff
@@ -0,0 +1,51 @@
+package no.unit.nva.publication.storage.model.daos;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import no.unit.nva.identifiers.SortableIdentifier;
+
+import java.util.Map;
+
+import static no.unit.nva.publication.storage.model.DatabaseConstants.CRISTIN_ID_INDEX_FIELD_PREFIX;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.KEY_FIELDS_DELIMITER;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCES_BY_CRISTIN_ID_INDEX_PARTITION_KEY_NAME;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCES_BY_CRISTIN_ID_INDEX_SORT_KEY_NAME;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCES_TABLE_NAME;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCE_BY_CRISTIN_ID_INDEX_NAME;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCE_INDEX_FIELD_PREFIX;
+
+public interface WithCristinIdentifier {
+
+    @JsonProperty(RESOURCES_BY_CRISTIN_ID_INDEX_PARTITION_KEY_NAME)
+    default String getResourceByCristinIdPartitionKey() {
+        return getCristinIdentifier() == null ? getCristinIdentifier()
+                : CRISTIN_ID_INDEX_FIELD_PREFIX + KEY_FIELDS_DELIMITER + getCristinIdentifier();
+    }
+
+    @JsonProperty(RESOURCES_BY_CRISTIN_ID_INDEX_SORT_KEY_NAME)
+    default String getResourceByCristinIdSortKey() {
+        return RESOURCE_INDEX_FIELD_PREFIX + KEY_FIELDS_DELIMITER + getIdentifier();
+    }
+
+    SortableIdentifier getIdentifier();
+
+    String getCristinIdentifier();
```

Comment:
Add @JsonIgnore in both getters. Add a test verifying that these values are not serialized.

Reply:
No idea how to test if something i serialized or not